barquesubmit/client.go

```diff
@@ -0,0 +1,203 @@
+	if err != nil {
+		return errors.Wrap(err, "problem building login request")
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "problem making login request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Errorf("login request returned %d", resp.StatusCode)
+	}
+
+	data := &userAPIKeyResponse{}
+	if err = gimlet.GetJSON(resp.Body, data); err != nil {
+		return errors.Wrap(err, "problem reading body of login response")
+	}
+
+	if data.Username != username {
+		return errors.Errorf("service returned logically inconsistent credentials")
+	}
+
+	c.apiKey = data.Key
+	return nil
+}
+
+func (c *Client) SubmitJob(ctx context.Context, opts repobuilder.JobOptions) (string, error) {
+	client := utility.GetDefaultHTTPRetryableClient()
+	defer utility.PutHTTPClient(client)
+
+	payload, err := json.Marshal(opts)
+	if err != nil {
+		return "", errors.Wrap(err, "problem marshaling json")
+	}
+
+	req, err := c.makeRequest(ctx, "repobuilder", http.MethodPost, bytes.NewBuffer(payload))
+	if err != nil {
+		return "", errors.Wrap(err, "problem building job request")
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", errors.Wrap(err, "problem making job submission request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", c.handleError(resp.Body)
+	}
+
+	out := struct {
+		ID     string   `json:"id"`
+		Scopes []string `json:"scopes"`
+	}{}
+
+	if err = gimlet.GetJSON(resp.Body, &out); err != nil {
+		return "", errors.Wrap(err, "problem reading body of login response")
+	}
+
+	return out.ID, nil
+}
+
+type JobStatus struct {
+	ID          string              `json:"id"`
+	Status      amboy.JobStatusInfo `json:"status"`
+	Timing      amboy.JobTimeInfo   `json:"timing"`
+	QueueStatus amboy.QueueStats    `json:"queue_status"`
+	HasErrors   bool                `json:"has_errors"`
+	Error       string              `json:"error"`
+}
+
+func (c *Client) handleError(body io.ReadCloser) gimlet.ErrorResponse {
+	out := gimlet.ErrorResponse{}
+	err := gimlet.GetJSON(body, &out)
+	if err != nil {
+		out.Message = errors.Wrap(err, "problem parsing error response").Error()
+		out.StatusCode = http.StatusUnprocessableEntity
+	}
+	return out
+}
+
+func (c *Client) CheckJobStatus(ctx context.Context, id string) (*JobStatus, error) {
+	client := utility.GetDefaultHTTPRetryableClient()
+	defer utility.PutHTTPClient(client)
+
+	req, err := c.makeRequest(ctx, path.Join("repobuilder", id), http.MethodGet, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem building job request")
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem making job submission request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, c.handleError(resp.Body)
+	}
+	out := &JobStatus{}
+	if err = gimlet.GetJSON(resp.Body, &out); err != nil {
```

Comment:
This doesn't need to take the address if you already have a pointer to JobStatus.